docs/explanations/architecture-2.md

```diff
@@ -0,0 +1,331 @@
+You can have different policies with the same Wasm module and settings
+running in many Policy Servers.
+However, you can't have a single policy definition that runs in many policy servers.
+
+:::
+
+The `kubewarden-controller` notices the new `ClusterAdmissionPolicy` resource and
+so finds the bound `PolicyServer` and reconciles it.
+
+### Reconciliation of `policy-server`
+
+When creating, modifying or deleting a `ClusterAdmissionPolicy` or `AdmissionPolicy`,
+a reconciliation loop activates in `kubewarden-controller`,
+for the `PolicyServer` owning the policy.
+This reconciliation loop creates a `ConfigMap` with all the polices bound to the `PolicyServer`.
+Then the Deployment rollout of the `policy-server` starts.
+It results in starting the new `policy-server` instance with the updated configuration.
+
+At start time, the `policy-server` reads its configuration from the ConfigMap
+and downloads all the Kubewarden policies specified.
+You can download Kubewarden policies from remote HTTP servers and container registries.
+
+You use policy settings parameters to tune a policies' behavior.
+After startup and policy download the `policy-server`
+checks the policy settings provided by the user are valid.
+
+The `policy-server` validates policy settings by invoking
+the `validate_setting` function exposed by each policy.
+There is further documentation in the
+[specification reference](../reference/spec/01-intro-spec.md)
+section of the documentation.
+
+If one or more policies received wrong configuration parameters,
+from the policy specification provided by the user,
+then any admission requests evaluated by that policy return an error.
+
+When Kubewarden has configured all policies,
+`policy-server`
+spawns a pool of worker threads to evaluate incoming requests
+using the Kubewarden policies specified by the user.
+
+Finally, the `policy-server` starts a HTTPS server,
+the Kubewarden Policy Server,
+listening to incoming validation requests.
+Kubewarden uses the TLS key and certificate
+created by `kubewarden-controller`
+to secure the web server.
+
+The web server exposes each policy by a dedicated path
+following the naming convention: `/validate/<policy ID>`.
+
+This diagram shows the cluster when initialization of `policy-server` is complete:
+
+### Making Kubernetes aware of the policy
+
+The `policy-server` Pods have a
+[`Readiness Probe`](https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/),
+that `kubewarden-controller` uses to check when
+the `policy-server` Deployment is ready to evaluate an
+[`AdmissionReview`](https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/#webhook-request-and-response).
+
+Once Kubewarden marks the `policy-server` Deployment as `Ready`,
+the `kubewarden-controller` makes the Kubernetes API server
+aware of the new policy by creating either a
+`MutatingWebhookConfiguration` or a `ValidatingWebhookConfiguration` object.
```

Comment:
Actually, the controller will create/update the webhooks when the policy is "uniquely reachable". Which means that all the policy server instances running in the cluster have the latest policy configuration running. That's necessary because due the rollout of of the policy servers works,  during the deploy of a policy is possible to have the same policy (one on each policy server) with two different configuration.

Reply:
I had a go, we can review in the next iteration.